src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker.cs

```diff
@@ -862,6 +890,27 @@ private void OutputToOutputWindow(string message)
             return (AbstractProject)projectContext;
         }
 
+        private Guid GetProjectGuid(string projectFilename)
+        {
+            var solution5 = (IVsSolution5)_vsSolution;
+            // If the index is stale, it might give us a path that doesn't exist anymore that the
+            // solution doesn't know about - be resilient to that case.
+            Guid projectGuid;
+            try
+            {
+                projectGuid = solution5.GetGuidOfProjectFile(projectFilename);
+            }
+            catch (ArgumentException)
+            {
+                var message = $"Failed to get the project guid for '{projectFilename}' from the solution, using  random guid instead.";
```

Comment:
Note that the output window stuff is `[Conditional("DEBUG")]` so it won't be there in dumps.